src/test/java/pages/HomePage.java

```diff
@@ -0,0 +1,265 @@
+
+    @FindBy (xpath = "//*[@id='songResultsWrapper']/header/div[3]/span/button[2]")
+    WebElement addToButton;
+
+
+    @FindBy (css = "div.success.show")
+    WebElement notificationMessage;
+
+    @FindBy (css = ".menu .songs")
+    WebElement allSongsButton;
+
+    @FindBy (css = "input[name='name']")
+    WebElement playlistTitleInput;
+
+
+
+    public HomePage updatePlaylistTitle(String newPlaylistTitle) {
+        wait.until(ExpectedConditions.visibilityOf(inputFieldWithTitle));
+        inputFieldWithTitle.sendKeys(Keys.chord(Keys.CONTROL, "a"));
+        inputFieldWithTitle.sendKeys(Keys.DELETE);
+        inputFieldWithTitle.sendKeys(newPlaylistTitle);
+        inputFieldWithTitle.sendKeys(Keys.ENTER);
+        return this;
+    }
+
+    public HomePage doubleClickOnPlaylist() {
+        wait.until(ExpectedConditions.visibilityOf(listToUpdate));
+        Actions actions = new Actions(driver);
+        actions.doubleClick(listToUpdate).perform();
+        return this;
+    }
+
+    public HomePage deleteCreatedPlaylist() {
+
+        wait.until(ExpectedConditions.visibilityOf(listToDelete));
+        listToDelete.click();
+        wait.until(ExpectedConditions.visibilityOf(deletePlaylistButton));
+        deletePlaylistButton.click();
+        wait.until(ExpectedConditions.visibilityOf(okButton));
+        okButton.click();
+        return this;
+    }
+
+    public HomePage searchForSong(String name) {
+        wait.until(ExpectedConditions.visibilityOf(searchField));
+        searchField.clear();
+        searchField.sendKeys(name);
+        return this;
+    }
+
+    public HomePage clickViewAllButton() {
+        wait.until(ExpectedConditions.visibilityOf(viewAllButton));
+        viewAllButton.click();
+        return this;
+    }
+
+    public HomePage findFirstSongInResults() {
+        wait.until(ExpectedConditions.visibilityOf(firstSongInList));
+        firstSongInList.click();
+        return this;
+    }
+
+    public HomePage clickAddToButton() {
+        wait.until(ExpectedConditions.visibilityOf(addToButton));
+        addToButton.click();
+        return this;
+    }
+
+    public HomePage addSongToMyNewPlaylist() {
+        wait.until(ExpectedConditions.visibilityOf(listButton));
+        Actions action = new Actions(driver);
+        action.moveToElement(listButton);
+        listButton.click();
+        return this;
+    }
+
+
+    public HomePage addSongToPlayList(String song) {
+        searchForSong(song);
+        clickViewAllButton();
+        findFirstSongInResults();
+        clickAddToButton();
+        addSongToMyNewPlaylist();
+        return this;
+    }
+
+    public HomePage createPlaylist(String myNewPlaylist) {
+        String newPlaylistButton = "#playlists .fa.fa-plus-circle.create";
+        clickOnElement(By.cssSelector(newPlaylistButton));
+
+        //String newPlaylistOption = ".menu.playlist-menu li:nth-child(1)";
+        clickOnElement(By.cssSelector(".menu.playlist-menu li:nth-child(1)"));
```

Comment:
But I have some suggestions below: in this `createPlaylist` function, you can use @FindBy annotation for declaring Web Elements. And at step click New Playlist button, you should define the locator for the element to make it easier to understand by using the element's attributes.
Example: ` `[data-testid*='create-simple']`` or `//*[text()='New Playlist']`

![image](https://github.com/Test-Pro-Bootcamp/A49-QA-Automation-Framework/assets/144403954/4467e242-51d6-4f4c-ada0-e8b8337c5c2a)